src/core/event_registry/server_event.rs

```diff
@@ -695,60 +657,204 @@ unsafe fn send_independent_event<E: Event>(
 
 /// Helper for serializing a server event.
 ///
-/// Will prepend the client's change tick to the injected message.
-/// Optimized to avoid reallocations when consecutive clients have the same change tick.
+/// Will prepend padding bytes for where the change tick will be inserted to the injected message.
 ///
 /// # Safety
 ///
 /// The caller must ensure that `event_data` was created for `E`.
-unsafe fn serialize_with_tick<E: Event>(
+unsafe fn serialize_with_padding<E: Event>(
     event_data: &ServerEvent,
     ctx: &mut ServerSendCtx,
     event: &E,
-    init_tick: RepliconTick,
-    previous_message: Option<SerializedMessage>,
 ) -> bincode::Result<SerializedMessage> {
-    if let Some(previous_message) = previous_message {
-        if previous_message.init_tick == init_tick {
-            return Ok(previous_message);
+    let bytes = Vec::with_capacity(128);
```

Comment:
I'm not sure about this... It's usually better to avoid pre-allocation using a random number without benchmarking.
Some events could be unit structs with zero-size, so only tick is necessary.

Maybe just use `Vec::new`? 
If we decode on this, I will be better to do the same for independent events as well.

Reply:
> It's usually better to avoid pre-allocation using a random number without benchmarking.

Afaik the cost of allocating once is much higher than the cost of different sized allocations. As long as we aren't preallocating multiple pages, it should be fine to overallocate a little. These events are short-lived so there shouldn't be memory constraints.

> Maybe just use Vec::new?

Vec::new almost guarantees spurious allocations for relatively small messages (the capacity [likely starts](https://stackoverflow.com/a/62686103) at 8 bytes). I picked 128 because it's a power of 2 that will give you 15 entities plus the tick plus some bools/chars. Enough for most statically-sized messages, anything bigger is most of the time going to be dynamically sized.

> If we decide on pre-allocation, it will be better to do the same for independent events as well.

Works for me.

Reply:
> Afaik the cost of allocating once is much higher than the cost of different sized allocations

That should be the case, yes. But 128 bytes is also quite big. I personally can't say for sure if it's faster without looking at benchmarks, they could surprise. 

> the capacity [likely starts](https://stackoverflow.com/a/62686103) at 8 bytes

Since it's enough to include the tick, maybe it's fine? Most events are zero-sized, or store a single entity / number serialized as varint.

This feels a bit like a premature optimization :thinking: `Vec::new` will be simpler then creating a constant, commenting it and passing into `with_capacity` without guarantied benefits.

Reply:
I will undo this so we can get the PR merged, and we can follow-up afterward. Ultimately I expect benchmarks won't reveal any new information, and we will just need to use a heuristic to pick a good starting capacity (we may be able to use `std::mem::size_of_val` to get a good approximation as well, although rust bytes don't necessarily line up 1:1 with serialized bytes).